saleor/plugins/webhook/tests/subscription_webhooks/test_create_deliveries_for_subscription.py

```diff
@@ -1644,3 +1866,30 @@ def test_generate_payload_from_subscription_return_permission_errors_in_payload(
     assert payload["errors"][0]["extensions"]["exception"]["code"] == error_code
     assert len(deliveries) == len(webhooks)
     assert deliveries[0].webhook == webhooks[0]
+
+
+def test_generate_sync_event_payload_with_circular_call(
+    checkout_ready_to_complete,
+    subscription_checkout_filter_shipping_methods_webhook_with_circular_fields,
+):
+
+    # given
+    webhooks = [
+        subscription_checkout_filter_shipping_methods_webhook_with_circular_fields
+    ]
+    event_type = WebhookEventSyncType.CHECKOUT_FILTER_SHIPPING_METHODS
```

Comment:
IMHO, each resolver that should raise such an exception should have a separate test that confirms that we handle this correctly. This test only checks that query with `checkout.shippingMethods` doesn't go into circular calls.

Reply:
Agree. Fixed